classes/webservice/WebserviceRequest.php

```diff
@@ -1542,7 +1548,23 @@ protected function saveEntityFromXml($successReturnCode)
 
             /* @var ObjectModel $object */
             if ($this->method == 'POST') {
-                $object = new $this->resourceConfiguration['retrieveData']['className']();
+                if ((string) $entity->id != '') {
+                    $object = new $this->resourceConfiguration['retrieveData']['className']((int) $attributes->id);
+                    if ($object->id) {
+                        $this->setError(400, 'id is already used', 92);
+
+                        return false;
+                    }
+                    if ((int) $attributes->force_id > 0) {
+                        $object->force_id = true;
+                    } else {
+                        $this->setError(400, 'id is forbidden when adding a new resource', 91);
```

Comment:
with the check on line 1537 it should never happen.
Sorry, it was written before.